roaring/src/bitmap/serialization.rs

```diff
@@ -47,6 +47,113 @@ impl RoaringBitmap {
         8 + container_sizes
     }
 
+    /// Creates a `RoaringBitmap` from a byte slice, interpreting the bytes as a bitmap with a specified offset.
+    ///
+    /// # Arguments
+    ///
+    /// - `offset: u32` - The starting position in the bitmap where the byte slice will be applied, specified in bits.
+    ///                   This means that if `offset` is `n`, the first byte in the slice will correspond to the `n`th bit(0-indexed) in the bitmap.
+    ///                   Must be a multiple of 8.
+    /// - `bytes: &[u8]` - The byte slice containing the bitmap data. The bytes are interpreted in "Least-Significant-First" bit order.
+    ///
+    /// # Interpretation of `bytes`
+    ///
+    /// The `bytes` slice is interpreted in "Least-Significant-First" bit order. Each byte is read from least significant bit (LSB) to most significant bit (MSB).
+    /// For example, the byte `0b00000101` represents the bits `1, 0, 1, 0, 0, 0, 0, 0` in that order (see Examples section).
+    ///
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if `offset` is not a multiple of 8, or if `bytes.len() + offset` is greater than 2^32.
+    ///
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use roaring::RoaringBitmap;
+    ///
+    /// let bytes = [0b00000101, 0b00000010, 0b00000000, 0b10000000];
+    /// //             ^^^^^^^^    ^^^^^^^^    ^^^^^^^^    ^^^^^^^^
+    /// //             76543210          98
+    /// let rb = RoaringBitmap::from_lsb0_bytes(0, &bytes);
+    /// assert!(rb.contains(0));
+    /// assert!(!rb.contains(1));
+    /// assert!(rb.contains(2));
+    /// assert!(rb.contains(9));
+    /// assert!(rb.contains(31));
+    ///
+    /// let rb = RoaringBitmap::from_lsb0_bytes(8, &bytes);
+    /// assert!(rb.contains(8));
+    /// assert!(!rb.contains(9));
+    /// assert!(rb.contains(10));
+    /// assert!(rb.contains(17));
+    /// assert!(rb.contains(39));
+    /// ```
+    pub fn from_lsb0_bytes(offset: u32, mut bytes: &[u8]) -> RoaringBitmap {
+        assert_eq!(offset % 8, 0, "offset must be a multiple of 8");
```

Comment:
Thank you for your review and sorry for the late responce. 

The reason why the offset must be 8 multiples is if the function accepts not 8 multiples offset, the copying will not so easy. If the container is Array, the implementation would be almost the same. In case Bitmap store, we need bit shifting for every bitmap byte copy.

We can switch the implementation based on the offset. In terms of the peformance, the users who use 8 multiples offset might not be impacted by this API change.
And the peformance of users with not 8 multiples offset is unknown.

I could allow any offset for this method with a little additional implementation and the document on peformances.

Reply:
Thank you, @lemolatoon, for the explanation. It would be great if we could accept for than just 8 multipliers, indeed.